lib/repl.js

```diff
@@ -157,12 +157,6 @@ function REPLServer(prompt,
         }
       } catch (e) {
         err = e;
-        if (err && process.domain) {
```

Comment:
This logic is moved to `finish` so all of the error handling can be in one place. It doesn't seem safe to do this here anyway since the evaluation function can be overriden. The explicit `process.domain.exit` is not carried over as 1) I don't understand what use it serves & I don't like to blindly c/p, 2) the other existing domain emit didn't come with an exit and 3) the repl+domain test passes without it. @isaacs you added this 2 years ago, do you remember?